confluent_kafka/avro/__init__.py

```diff
@@ -81,6 +81,45 @@ def produce(self, **kwargs):
         super(AvroProducer, self).produce(topic, value, key, **kwargs)
 
 
+class AvroMessage(object):
```

Comment:
This actually becomes fairly expensive on a per message basis. Rather than pulling the message type into python I'd suggest simply setting the additional attributes at runtime. 

~~~
def decode_message(self, message, schema_attr=None):
...
if not schema_attr is None: 
    msg.setattr(schema_attr, key_schema_obj)
...
~~~

This should be fairly cheap although admittedly it takes away from the readability a bit. Proper documentation should be able to bridge the gap here though.

Reply:
Actually it looks like the Message type fails to set the slot for `__setattr__`, standby

Reply:
yeah... that's why i had to go this route.  i'm open to changing it if you have a better plan though.

Reply:
@rnpridgeon do you have updates on approach for this?